stac_fastapi/pgstac/core.py

```diff
@@ -86,9 +85,15 @@ async def all_collections(  # noqa: C901
                 sortby=sortby,
                 filter_query=filter_expr,
                 filter_lang=filter_lang,
-                q=q,
+                **kwargs,
             )
 
+            # NOTE: `FreeTextExtension` - pgstac will only accept `str` so we need to
+            # join the list[str] with ` OR `
+            # ref: https://github.com/stac-utils/stac-fastapi-pgstac/pull/263
+            if q := clean_args.pop("q", None):
+                clean_args["q"] = " OR ".join(q) if isinstance(q, list) else q
```

Comment:
We need custom code to handle `list[str]` passed by collection-search Free-Text extension as pgstac will only accept `str` 

Note: we don't need this in items search because we will use pydantic serialization